boards/PDM/Inc/App/App_InRangeCheck.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+struct InRangeCheck;
+
+enum InRangeCheck_Status
+{
+    VALUE_IN_RANGE,
+    VALUE_UNDERFLOW,
+    VALUE_OVERFLOW,
+};
+
+/**
+ * Allocate and initialize an in-range check to check whether a value is in
+ * the given min/max range
+ * @get_voltage A function that can be called to get the value
+ * @min_value Minimum value in the range, inclusive
+ * @max_value Maximum value in the range, inclusive
+ * @error_callback The function to call when the value is out of the specified
+ *                 range
+ * @return A pointer to the created in-range check, whose ownership is given
+ * to the caller
+ */
+struct InRangeCheck *App_InRangeCheck_Create(
```

Comment:
Replaced voltage monitor with a more generic `InRangeCheck` that can be used for current too